controller/src/beerocks/master/tasks/dynamic_channel_selection_task.cpp

```diff
@@ -194,16 +190,13 @@ void dynamic_channel_selection_task::work()
     case eState::SCAN_DONE: {
         LOG(TRACE) << "SCAN_DONE";
 
-        if (m_scanning_task_id != id) {
+        if (!finish_scan()) {
```

Comment:
A better solution may be to add a new `FINISH` state, where you finalize everything.
Then you won't have to do this from multiple states.

Reply:
I agree - but this is more of refactoring the scan_done and abort_scan cleanups to be in a new state - and less of the actual add of protection.

I can add this as part of this PR.
@kantera800 what do you think?

Reply:
good idea, steering task also use such a state